### PR TITLE
CI pipeline for web-twig package

### PR DESCRIPTION
**Pull request by literat**

### CODE REVIEW

**Comment by literat** [comment]
- [ ] Merge https://github.com/lmc-eu/spirit-design-system/pull/650
- [ ] Merge https://github.com/lmc-eu/spirit-design-system/pull/651

**Review by adamkudrna — COMMENTED** [review]
Nice, looks promising! I'll keep an eye on that, there are obviously things yet to be fixed 🙂.

**Comment by literat** [comment]
> Nice, looks promising! I'll keep an eye on that, there are obviously things yet to be fixed 🙂.

@adamkudrna Ready to re-review.

**Review by tomassychra — APPROVED** [review]

**Comment by literat** [comment]
There is a potential problem in twigx-bundle:3.2.0. Version 3.1.0 works fine.

**Comment by literat** [comment]
Problem is in negative lookahead.

File: .github/workflows/test-web-twig.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+name: 'Web Twig: Code Quality Checks'
+
+on:
+  push:
+    paths:
+      - packages/web-twig/**
+      # Because of snapshot testing
```
Comment (adamkudrna) [inline .github/workflows/test-web-twig.yaml:7]: The comment could be a little bit more helpful like this, I believe:

```suggestion
      # Icons are part of `web-twig` snapshots, check if they are up-to-date.
```

Or something like that, YKWIM.

File: .github/workflows/test-web-twig.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      - name: Submit coverage to Coveralls
+        working-directory: ./packages/web-twig
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COVERALLS_PARALLEL: true
+          COVERALLS_FLAG_NAME: ${{ github.job }}-PHP-${{ matrix.php-version }} ${{ matrix.dependencies }}
+        run: |
+          composer global require php-coveralls/php-coveralls
+          ~/.composer/vendor/bin/php-coveralls --coverage_clover=coverage/clover.xml --json_path=coverage/coveralls-upload.json -v
+  finish-tests:
```
Comment (adamkudrna) [inline .github/workflows/test-web-twig.yaml:61]: Not sure, but how about _finalize_…?

**Review by adamkudrna — COMMENTED** [review]
There are still some required checks failing, but it looks promising!

File: .github/workflows/test-web-twig.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
+name: 'Web Twig: Code Quality Checks'
+
+on:
+  push:
+    paths:
+      - packages/web-twig/**
+      # Changes in SVG icons also affects snapshots in web-twig package
```
Comment (adamkudrna) [inline .github/workflows/test-web-twig.yaml:7]:
```suggestion
      # Changes in SVG icons also affect snapshots in web-twig package
```

**Review by adamkudrna — COMMENTED** [review]
As agreed in person, please streamline the workflow so we get what matters most, i.e. the snapshot tests on any environment:

- just a single version of PHP is OK for now,
- coverage publishing is also not necessary, should it cause any issues.

Also, I think the `test` category suits better than `maintenance` for this kind of PR, what do you think?

File: .github/workflows/test-web-twig.yaml (hunk 3 above)
Comment (literat) [reply .github/workflows/test-web-twig.yaml]: Fixed.

**Comment by literat** [comment]
> Also, I think the `test` category suits better than `maintenance` for this kind of PR, what do you think?

I do not understand. There is no `test` label. Or you are talking about the commit message type?

**Comment by literat** [comment]
No this is working after fixing bugs in TwigX.

File: .github/workflows/test-web-twig.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  codestyle:
+    name: 'Code style and static analysis'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
```
Comment (crishpeen) [inline .github/workflows/test-web-twig.yaml:87]: here we have 8.1? Shouldn't there be the same PHP version everywhere?

File: .github/workflows/test-web-twig.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# TODO: Enable matrix and other code after the problem with negative lookahead in TwigX is solved
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/test-web-twig.yaml:1]: 🤦 This is solved so I can uncomment :-)

File: .github/workflows/test-web-twig.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      - packages/icons/**
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        # php-version: ['7.4', '8.1', '8.2']
+        php-version: ['7.4']
```
[the remark this answers is not in the document]
Comment (literat) [reply .github/workflows/test-web-twig.yaml]: Fixed.

File: .github/workflows/test-web-twig.yaml (hunk 4 above)
Comment (literat) [reply .github/workflows/test-web-twig.yaml:86]: I have returned to the matrix.

**Pull request merged**